addons/point_of_sale/static/src/js/Screens/PaymentScreen/PaymentScreen.js

```diff
@@ -94,6 +99,34 @@ odoo.define('point_of_sale.PaymentScreen', function (require) {
                 return false;
             }
         }
+        _display_popup_error_paymentlines_rounding() {
+            if(this.env.pos.config.cash_rounding) {
+                var orderlines = this.env.pos.get_order().get_paymentlines();
+                var cash_rounding = this.env.pos.cash_rounding[0].rounding;
+                var default_rounding = this.env.pos.currency.rounding;
+                for(var id in orderlines) {
+                    var line = orderlines[id];
+                    var diff = round_pr(round_pr(line.amount, cash_rounding) - round_pr(line.amount, default_rounding), default_rounding);
+
+                    if(diff && (line.payment_method.is_cash_count || !this.env.pos.config.only_round_cash_method)) {
+                        this.showPopup("ErrorPopup", {
+                            title: this.env._t("Rounding error in payment lines"),
+                            body: _.str.sprintf(
+                                this.env._t(
+                                    "The amount of your payment lines must be rounded to validate the transaction.\n" +
+                                    "The rounding precision is %s so you should set %s or %s as payment amount instead of %s."
+                                ),
+                                cash_rounding.toFixed(this.env.pos.currency.decimal_places),
+                                (round_pr(line.amount, default_rounding) - diff).toFixed(this.env.pos.currency.decimal_places),
+                                (round_pr(line.amount, default_rounding) + diff).toFixed(this.env.pos.currency.decimal_places),
+                                line.amount.toFixed(this.env.pos.currency.decimal_places)
```

Comment:
This is wrong if the diff is not the same in both directions (example rounding 0.1 and value is 1.24 -> text give me 1.20 or 1.28 which appears to not be true). Compute the two different diff